packages/core/src/types.ts

```diff
@@ -894,7 +932,15 @@ export enum ActionTypes {
   Choose = 'xstate.choose'
 }
 
-export interface RaiseAction<TEvent extends EventObject> {
+// Used as a "phantom type" for built-in actions
+export interface BuiltInActionObject<TContext, TEvent extends EventObject> {
+  type: string;
+  __context: TContext;
+  __event: TEvent;
```

Comment:
you could make those optional to avoid casting return values in the action resolvers: 
https://github.com/statelyai/xstate/commit/8ca10ac7e56daf8bd4d5355d51f1ec5535a01a4f

Reply:
Hm, it turns out that this doesn't report those are problems anymore:
https://github.com/statelyai/xstate/blob/6cf01b4cf8dbf26b266674de1fa415e43b248bcc/packages/core/test/schema.test.ts#L110-L113

It seems highly related to this comment: https://github.com/statelyai/xstate/pull/2518#issuecomment-903603539 . It's because both builtin action objects and `TAction` use `type: string`, so those types are overlapping - where in reality they are 2 different concepts